3b.run_sam/scripts/1.run_sam2_microscopy.py

```diff
@@ -0,0 +1,615 @@
+
+
+downscale_factor = 12
+# sort the files by name
+# downsample the image
+for f in files:
+    img = io.imread(f)
+    # downsample the image
+    downsampled_img = img[::downscale_factor, ::downscale_factor]
+    # save the downsampled image in place of the original image
+    io.imsave(f, downsampled_img)
+
+
+# ## 3. Get initial masks and centers via StarDist
+
+# ### Get the first frame of each video
+# ### Set up a dict that holds the images path, the first frame_mask, and the first frame_centers
+
+# In[ ]:
+
+
+# where one image set here is a single well and fov over all timepoints
+image_set_dict = {
+    "image_set_name": [],
+    "image_set_path": [],
+    "image_set_first_frame": [],
+    "image_x_y_coords": [],
+    "image_labels": [],
+}
+
+# get the list of directories in the ordered tiffs directory
+dirs = list(converted_to_video_dir.glob("*"))
+dirs = [dir for dir in dirs if dir.is_dir()]
+for dir in dirs:
+    # get the files in the directory
+    files = sorted(dir.glob("*.jpeg"))
+    image_set_dict["image_set_name"].append(dir.name)
+    image_set_dict["image_set_path"].append(str(dir))
+    image_set_dict["image_set_first_frame"].append(files[0])
+
+
+# ### Plot the segementation
+# Plot the following:
+# - the original image
+# - the segmentation
+# - the x,y centers of the segmentation
+# - the extracted masks
+
+# In[ ]:
+
+
+model = StarDist2D.from_pretrained("2D_versatile_fluo")
+
+# loop through each image set and predict the instances
+for i in range(len(image_set_dict["image_set_name"])):
+    print(
+        f"{image_set_dict['image_set_name'][i]}: {image_set_dict['image_set_first_frame'][i]}"
+    )
+    img = io.imread(image_set_dict["image_set_first_frame"][i])
+    labels, _ = model.predict_instances(normalize(img))
+    # convert the labels into position coordinates
+    regions = regionprops(label(labels))
+    coords = np.array([r.centroid for r in regions])
+
+    # plot the points and the masks and the image side by side by side
+    fig, ax = plt.subplots(1, 4, figsize=(30, 15))
+    ax[0].imshow(img, cmap="gray")
+    ax[0].set_title("Image")
+    ax[1].imshow(render_label(labels, img=img))
+    ax[1].set_title("Masks")
+    ax[2].imshow(img, cmap="gray")
+    ax[2].scatter(
+        coords[:, 1],
+        coords[:, 0],
+        color="red",
+        marker="*",
+        s=100,
+        edgecolor="white",
+        linewidth=1.25,
+    )
+    ax[2].set_title("Points")
+    coords = coords[:, [1, 0]]
+    ax[3].invert_yaxis()
+    # make the aspect ratio equal
+    ax[3].set_aspect("equal")
+    show_points(coords, np.ones(len(coords)), ax[3])
+    labels = np.ones(coords.shape[0], dtype=np.int32)
+    image_set_dict["image_x_y_coords"].append(coords)
+    image_set_dict["image_labels"].append(labels)
+
+# remove star dist model from memory
+del model
+# remove all stardist gpu memory
+torch.cuda.empty_cache()
+
+
+# ## 4. Track multiple objects in the video
+
+# In[ ]:
+
+
+image_set_dict.keys()
```

Comment:
Recommend adding a code comment for this part since you are printing out the keys I believe?

Reply:
removed this cell